crates/core/tedge_agent/src/entity_manager/tests.rs

```diff
@@ -0,0 +1,542 @@
+use crate::entity_manager::server::EntityStoreResponse;
+use proptest::proptest;
+use std::collections::HashSet;
+use tedge_actors::Server;
+use tedge_api::entity::EntityMetadata;
+
+#[tokio::test]
+async fn new_entity_store() {
+    let (mut entity_store, _mqtt_output) = entity::server("device-under-test");
+
+    assert_eq!(
+        entity::get(&mut entity_store, "device/main//").await,
+        Some(EntityMetadata::main_device("device-under-test".to_string()))
+    )
```

Comment:
This test is failing and I will fixed it. However, I would like to understand why the method `EntityRegistrationMessage::main_device()` takes an optional external id for the device but ignores this value.

Reply:
My bad. Should have removed that argument after moving the external id aspects from the entity store to the entity cache.

Reply:
Okay, I will do that in this PR.

Reply:
Done: https://github.com/thin-edge/thin-edge.io/pull/3344/commits/2ae55c313cde87896695f866baeb3fd6c126b95a